From 7ca699fc17c2a57ed13662df42eb9f52524f2f63 Mon Sep 17 00:00:00 2001
From: "guoyin.chen" <guoyin.chen@freescale.com>
Date: Wed, 24 Jul 2013 18:07:01 +0800
Subject: [PATCH 08/34] ENGR00272459 Make sure kernel will be rebuilt for
 different board config

Change the dependency kernelimage target to $(PRODUCT_OUT)/.config from
kernel_imx/.config, which will make sure different board config have
different $(PRODUCT_OUT)/.config.

Signed-off-by: guoyin.chen <guoyin.chen@freescale.com>
---
 core/Makefile | 11 ++++-------
 1 file changed, 4 insertions(+), 7 deletions(-)

diff --git a/core/Makefile b/core/Makefile
index 98a4543..84973f8 100644
--- a/core/Makefile
+++ b/core/Makefile
@@ -977,23 +977,20 @@ HOST_PROCESSOR := $(shell cat /proc/cpuinfo | grep processor | wc -l)
 INSTALLED_KERNEL_TARGET := $(PRODUCT_OUT)/kernel
 TARGET_PREBUILT_KERNEL := $(PRODUCT_OUT)/kernel
 KERNEL_CONFIGURE := kernel_imx/.config
+TARGET_KERNEL_CONFIGURE := $(PRODUCT_OUT)/.config
 KERNEL_ZIMAGE    := kernel_imx/arch/arm/boot/zImage
 KERNEL_UIMAGE    := kernel_imx/arch/arm/boot/uImage
 KERNEL_OUT := $(TARGET_OUT_INTERMEDIATES)/KERNEL_OBJ
-KERNEL_HEADERS_INSTALL := $(KERNEL_OUT)/usr
 
 
 .PHONY: $(TARGET_PREBUILT_KERNEL)
 kernelimage: $(TARGET_PREBUILT_KERNEL)
 
-$(KERNEL_CONFIGURE): kernel_imx/arch/arm/configs/$(TARGET_KERNEL_DEFCONF)
+$(TARGET_KERNEL_CONFIGURE): kernel_imx/arch/arm/configs/$(TARGET_KERNEL_DEFCONF) $(KERNEL_CONFIGURE)
 	$(MAKE) -C kernel_imx $(TARGET_KERNEL_DEFCONF) $(KERNEL_ENV)
+	install -D $(KERNEL_CONFIGURE) $(TARGET_KERNEL_CONFIGURE)
 
-$(KERNEL_HEADERS_INSTALL): $(KERNEL_CONFIGURE)
-	install -d $(KERNEL_OUT)
-	$(MAKE) -C kernel_imx -j$(HOST_PROCESSOR) O=../$(KERNEL_OUT) $(KERNEL_ENV) headers_install
-
-$(TARGET_PREBUILT_KERNEL): $(KERNEL_CONFIGURE)
+$(TARGET_PREBUILT_KERNEL): $(TARGET_KERNEL_CONFIGURE)
 	$(MAKE) -C kernel_imx -j$(HOST_PROCESSOR) uImage $(KERNEL_ENV)
 	install -D $(KERNEL_UIMAGE)  $(PRODUCT_OUT)/uImage
 	install -D $(KERNEL_ZIMAGE)  $(PRODUCT_OUT)/kernel
-- 
1.8.0

